README.md

```diff
@@ -4,3 +4,198 @@ Application-friendly RethinkDB client
 
 [![Build Status](https://secure.travis-ci.org/hueniverse/penseur.png)](http://travis-ci.org/hueniverse/penseur)
 
+
+## API
+
+### `new Penseur.Db(name, [options])`
+
+Creates a new `Db` object where:
+- `name` - name of the database to use. Default is 'test'.
+- `options` - optional configuration:
+  - `host` - string containing the host to connect to. Default is 'localhost'
+  - `port` - port to connect on. Default is 28015.
+  - `authKey` - string containing an established authorization key
+  - `user` - string containing the user you want to perform actions as. Default is 'admin'
+  - `password` - string containing the users password
+  - `timeout` - number indicating the number of seconds to wait for a connection to be established before timing out. Default is 20 seconds.
+  - `ssl` - ssl settings when RethinkDB requires a secure connection. This setting is passed to the RethinkDB module.
+    - `ca` - certificate as a Buffer used for establishing a secure connection to rethink, see [RethinkDB connect documentation](https://www.rethinkdb.com/api/javascript/connect/)
+  - `test` - boolean indicating if the `Db` actions should take place. Useful for testing scenarios, do not use in production.
+  - `extended` - object that is used to extend the members of the `Db` object.
+  - `onConnect` - function to execute when a connection is established.
+  - `onError` - function to execute when an error occurs. Signature is `(err)` where `err` is an error object.
+  - `onDisconnect`- function to execute when a disconnection occurs. Signature is `(willReconnect)` where `willReconnect` is a boolean indicating if the connection will be reestablished.
+  - `reconnect` - boolean indicating if the connection should be reestablished when interrupted
+  - `reconnectTimeout` - number of milliseconds to wait between reconnect attempts, when `false` there is no timeout. Defaults to `100`
+
+
+#### `db.connect(callback)`
+
+Create a connection to the database. `callback` is a function with the signature `(err)` that is executed when the connection is established or an error occurs establishing the connection.
+
+
+#### `db.close([next])`
+
+Close all database connections.
+
+
+#### `db.establish([tables], callback)`
+
+Establish a connection if one doesn't exist and create the database and tables if they don't already exist. This function also decorates the `db` object with helper properties to for using each of the tables.
+
```

Comment:
A warning is probably warranted here since this is a write-enabled method. It can change your data, set indexes, create new tables... It is meant for testing and scripts, not for actual production runtime.